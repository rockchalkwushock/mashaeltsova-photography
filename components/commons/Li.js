import PropTypes from 'prop-types'

const Li = ({ text }) => (
  <li>
    {text}
    <style jsx>{`
      li {
        padding: 0.25em 0;
        text-transform: uppercase;
      }
    `}</style>
  </li>
)

Li.propTypes = {
  text: PropTypes.string.isRequired
}

export default Li
