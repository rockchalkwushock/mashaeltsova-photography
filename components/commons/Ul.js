const Ul = ({ children }) => (
  <ul>
    {children}
    <style jsx>{`
      ul {
        flex: 1 1 auto;
        padding-top: 1em;
        text-align: center;
      }
    `}</style>
  </ul>
)

export default Ul
