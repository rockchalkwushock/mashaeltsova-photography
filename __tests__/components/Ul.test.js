import React from 'react'
import { render } from 'enzyme'

import Ul from '../../components/commons/Ul'
import Li from '../../components/commons/Li'

describe('Component: Ul', () => {
  test('Renders without exploding', () => {
    const tree = render(
      <Ul>
        <Li text="Hello World!" />
        <Li text="What is crackalacking?" />
        <Li text="Boooooooooooo!" />
      </Ul>
    )
    expect(tree).toMatchSnapshot()
  })
})
