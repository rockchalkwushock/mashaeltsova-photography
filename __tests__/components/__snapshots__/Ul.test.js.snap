// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Ul Renders without exploding 1`] = `
<ul
  class="jsx-2686926528"
>
  <li
    class="jsx-2559750438"
  >
    Hello World!
  </li>
  <li
    class="jsx-2559750438"
  >
    What is crackalacking?
  </li>
  <li
    class="jsx-2559750438"
  >
    Boooooooooooo!
  </li>
</ul>
`;
